datafusion/functions/src/datetime/make_interval.rs

```diff
@@ -0,0 +1,605 @@
+                plan_datafusion_err!("make_dt_interval arg[2] must be Int32")
+            })
+        })
+        .transpose()?;
+    let days: Option<&PrimitiveArray<Int32Type>> = args
+        .get(3)
+        .map(|a| {
+            a.as_primitive_opt::<Int32Type>().ok_or_else(|| {
+                plan_datafusion_err!("make_dt_interval arg[3] must be Int32")
+            })
+        })
+        .transpose()?;
+    let hours: Option<&PrimitiveArray<Int32Type>> = args
+        .get(4)
+        .map(|a| {
+            a.as_primitive_opt::<Int32Type>().ok_or_else(|| {
+                plan_datafusion_err!("make_dt_interval arg[4] must be Int32")
+            })
+        })
+        .transpose()?;
+    let mins: Option<&PrimitiveArray<Int32Type>> = args
+        .get(5)
+        .map(|a| {
+            a.as_primitive_opt::<Int32Type>().ok_or_else(|| {
+                plan_datafusion_err!("make_dt_interval arg[5] must be Int32")
+            })
+        })
+        .transpose()?;
+    let secs: Option<&PrimitiveArray<Float64Type>> = args
+        .get(6)
+        .map(|a| {
+            a.as_primitive_opt::<Float64Type>().ok_or_else(|| {
+                plan_datafusion_err!("make_dt_interval arg[6] must be Float64")
+            })
+        })
+        .transpose()?;
+
+    let mut builder = IntervalMonthDayNanoBuilder::with_capacity(n_rows);
+
+    for i in 0..n_rows {
+        // if one column is NULL → result NULL
+        let any_null_present = years.is_null(i)
+            || months.as_ref().is_some_and(|a| a.is_null(i))
+            || weeks.as_ref().is_some_and(|a| a.is_null(i))
+            || days.as_ref().is_some_and(|a| a.is_null(i))
+            || hours.as_ref().is_some_and(|a| a.is_null(i))
+            || mins.as_ref().is_some_and(|a| a.is_null(i))
+            || secs.as_ref().is_some_and(|a| {
+                a.is_null(i) || !a.value(i).is_finite() || a.value(i).is_nan()
+            });
+
+        if any_null_present {
+            builder.append_null();
+            continue;
+        }
+
+        // default values 0 or 0.0
+        let y = years.value(i);
+        let mo = months.as_ref().map_or(0, |a| a.value(i));
+        let w = weeks.as_ref().map_or(0, |a| a.value(i));
+        let d = days.as_ref().map_or(0, |a| a.value(i));
+        let h = hours.as_ref().map_or(0, |a| a.value(i));
+        let mi = mins.as_ref().map_or(0, |a| a.value(i));
+        let s = secs.as_ref().map_or(0.0, |a| a.value(i));
+
+        match make_interval_month_day_nano(y, mo, w, d, h, mi, s)? {
+            Some(v) => builder.append_value(v),
+            None => {
+                builder.append_null();
+                continue;
+            }
+        }
+    }
+
+    Ok(Arc::new(builder.finish()))
+}
+
+pub fn make_interval_month_day_nano(
+    year: i32,
+    month: i32,
+    week: i32,
+    day: i32,
+    hour: i32,
+    min: i32,
+    sec: f64,
+) -> Result<Option<IntervalMonthDayNano>> {
```

Comment:
```suggestion
) -> Option<IntervalMonthDayNano> {
```

Now that the overflow -> error flow is gone, can simplify this.